### PR TITLE
Add dynamic library for libvmaf.

### PR DESCRIPTION
**Pull request by christosbampis**

### CODE REVIEW

**Comment by kylophone** [comment]
I have another patch coming later today which will also provide a dylib. What does this test check for, linkability?

**Comment by christosbampis** [comment]
Yeah, it checks for linkability for the dynamic library. We can cherry pick that test file.

File: src/libvmaf/Makefile [hunk 1]
```diff
@@ -1,8 +1,10 @@
 LIBVMAF = libvmaf.a
+LIBVMAF_DYN = libvmafdyn.so
```
Comment (li-zhi) [inline src/libvmaf/Makefile]: could we just call it `libvmaf.so`?

Comment (christosbampis) [reply src/libvmaf/Makefile]: Yeah, we can rename if we want to. Just gave this name for simplicity to avoid confusion with libvmaf.a when doing -lvmaf.

Comment (kylophone) [reply src/libvmaf/Makefile]: The suffix for these is platform dependent. `.dylib` on MacOS or `.dll` on Windows for example.

See line 3b here: http://web.archive.org/web/20110720003722/http://www.fortran-2000.com/ArnaudRecipes/sharedlib.html

It is better to use a build system like `CMake` or `meson` to handle this. I will be sending a PR soon to address this.

Comment (christosbampis) [reply src/libvmaf/Makefile]: sure thing, that will help make building cleaner!

**Pull request merged**